Description: retrieve translations with the X-Ubuntu-Gettext-Domain key
 Display translations for patched desktop files from Ubuntu whose
 translations are stored in a different place ("X-Ubuntu-Gettext-Domain"
 key is used to find it).
Author: Lionel Le Folgoc <mrpouit@ubuntu.com>
Forwarded: not-needed

--- libxfce4util-4.6.0.orig/libxfce4util/xfce-rc.c
+++ libxfce4util-4.6.0/libxfce4util/xfce-rc.c
@@ -506,11 +506,25 @@
                     const gchar  *fallback)
 {
   const gchar *value;
+  gboolean     has_gettext_domain;
+  const gchar *gettext_domain;
 
   g_return_val_if_fail (rc != NULL, fallback);
   g_return_val_if_fail (rc->read_entry != NULL, fallback);
   g_return_val_if_fail (key != NULL, fallback);
 
+  has_gettext_domain = xfce_rc_has_entry (rc, "X-Ubuntu-Gettext-Domain");
+
+  if (has_gettext_domain == TRUE && (strcmp (key, "Name") == 0 || strcmp (key, "GenericName") == 0 || strcmp (key, "Comment") == 0))
+  {
+    gettext_domain = (*rc->read_entry) (rc, "X-Ubuntu-Gettext-Domain", FALSE);
+
+    if (gettext_domain != NULL)
+    {
+      value = dgettext (gettext_domain, xfce_rc_read_entry_untranslated (rc, key, fallback));
+      return value;
+    }
+  }
   value = (*rc->read_entry) (rc, key, TRUE);
   if (value == NULL)
     value = fallback;
